src/pages/contract/_components/ReadContract.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  CursorPage,
+  CursorPagination,
+} from "@components/commons/CursorPagination";
+import { useRouter } from "next/router";
+import ReadContractCodeFile from "./ReadContractCodeFile";
+import VerifiedContractSubtitle from "./VerifiedContractSubtitle";
+
+export interface CodesData {
+  fileName: string;
+  code: string;
+}
+
+export default function ReadContract({
+  contractName,
+  compilerVersion,
+  evmVersion,
+  optimizedEnabled,
+  optimizationRuns,
+  verifiedAt,
+  codes,
+  pages,
+}: {
+  contractName: string;
+  compilerVersion: string;
+  evmVersion: string;
+  optimizedEnabled: boolean;
+  optimizationRuns: number;
+  verifiedAt: string;
+  codes: CodesData[];
+  pages: CursorPage[];
+}): JSX.Element {
+  const router = useRouter();
+  const id = router.query.cid;
+  return (
+    <>
+      <div className="text-white-50 font-bold text-xl mt-[42.5px] mb-[30.5px]">
+        {fixedTitle.codeDetails}
+      </div>
+      {/* desktop */}
+      <div className="hidden lg:block">
```

Comment:
too many repeated codes, let's use this [hook](https://github.com/WavesHQ/bridge/blob/main/apps/web/src/hooks/useResponsive.ts) and apply conditional styling

Reply:
to be work on future PR